app/views/users/_form.html.haml

```diff
@@ -1,9 +1,11 @@
 = simple_form_for [current_facility, @user] do |f|
   .form-inputs
-    = f.input :first_name
-    = f.input :last_name
-    = f.input :email
+    = f.input :first_name, readonly: @is_edit && !@user.admin_editable?
```

Comment:
I am fully aware that this is extremely gross. Can you suggest a better way of handling this?